fix(actor): Configured the delete operation in the state provider

ruokun-niu opened the pull request:
# Description

There was a bug in the `saveState()` function in the StateProvider. For the states that are marked to be removed, the correct operation should be `delete` instead of `remove`. This bug caused the deletion of an dapr actor state to be unsuccessful.

## Issue reference


Please reference the issue this PR will close: #484 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [ ] Code compiles correctly
- [ ] Created/updated tests
- [ ] Extended the documentation

XavierGeerinck commented:
Hi there! Thank you for this PR, would it be possible to let a test reflect this change so it won't break in the future?

shubham1172 reviewed (changes requested):
+1 to @XavierGeerinck's comments

ruokun-niu commented:
Yep for sure! I'll add a test today

ruokun-niu commented:
Hi @shubham1172 @XavierGeerinck I have just added a test for the stuff I implemented. Please feel free to review them and provide any feedbacks/criticism (It's my first time contributing to this repo so I might've implemented the tests incorrectly)

shubham1172 commented:
Hi @ruokun-niu, were you able to run this test locally? It seems to be failing in the CI, could you please have a look?

<pre>
deleteActorState
== APP ==       ✕ should be able to delete actor state (83 ms)
</pre>

shubham1172 commented on test/e2e/http/actors.test.ts:
```diff
@@ -84,6 +86,7 @@ describe("http/actors", () => {
     await server.actor.registerActor(DemoActorActivateImpl);
     await server.actor.registerActor(DemoActorTimerTtlImpl);
     await server.actor.registerActor(DemoActorReminderTtlImpl);
+    await server.actor.registerActor(DemoActorDeleteStateImpl);
```
Can you please line 112 to update the number of registered actors? It should be 9 after this change.

ruokun-niu commented:
Hi @shubham1172 , I am regularly getting the following error `initialization error occurred for pubsub-mqtt (pubsub.mqtt/v1): failed to establish connection to broker: context deadline exceeded`. Got any idea as to why this is being thrown

ruokun-niu commented:
Hi @shubham1172  sorry for the delayed response but I have fixed the failed test. Have also tested it locally and all of the tests were passing

ruokun-niu commented:
@shubham1172  I don't think I have access to merge this PR. Can you merge it for me? Thanks a lot :)

XavierGeerinck commented:
Thank you @ruokun-niu !
